examples/testify/src/integration-test/groovy/gorm/tools/repository/HasChangedIntegrationSpec.groovy

```diff
@@ -0,0 +1,266 @@
+package gorm.tools.repository
+
+import grails.gorm.transactions.Rollback
+import grails.testing.mixin.integration.Integration
+import spock.lang.Specification
+import yakworks.rally.orgs.model.Org
+import yakworks.rally.orgs.model.OrgType
+import yakworks.testing.gorm.integration.DomainIntTest
+import yakworks.testing.gorm.model.KitchenSink
+
+import java.time.LocalDateTime
+
+import yakworks.testing.gorm.model.KitchenSinkRepo
+import yakworks.testing.gorm.model.SinkExt
+import yakworks.testing.gorm.model.Thing
+
+/**
+ * COPY OF HasChangedSpec, sanity check same thing with integration tests
+ */
+@Integration
+@Rollback
+class HasChangedIntegrationSpec extends Specification implements DomainIntTest {
+
+    def setupSpec() {
+        KitchenSinkRepo.doTestAuditStamp = false
+    }
+
+    void "Org smoke test"() {
+        when:
+        Org org = new Org(num:"o-1", name:"o-1", companyId: 2, type: OrgType.Customer).persist()
+        org.persist()
+
+        then:
+        !org.hasChanged("num")
+        !org.hasChanged()
+
+    }
+
+    void "hasChanged on new"() {
+        when:
+        KitchenSink sink = new KitchenSink(num: "123", name: "name")
+
+        then: "new objects show up as changed"
+        sink.hasChanged()
+        sink.hasChanged("num")
+
+        and: "but show even if it didnt actually get populated, so its not accurate on fields"
+        sink.hasChanged("name2")
+    }
+
+    void "hasChanged change persisted"() {
+        when:
+        KitchenSink sink = new KitchenSink(num: "123", name: "name").persist()
+
+        then:
+        !sink.hasChanged()
+
+        when:
+        //change trait field
+        sink.num = "456"
+        //change KitchenSink field
+        sink.name2 = "foo"
+
+        then:
+        sink.hasChanged("num")
+        sink.hasChanged("name2")
+        sink.hasChanged()
+
+    }
+
+    void "markDirty works"() {
+        when:
+        KitchenSink sink = new KitchenSink(num: "123", name: "name").persist()
+        sink.markDirty()
+
+        then:
+        sink.hasChanged()
+
+        when:
+        sink.num = "456"
+
+        then:
+        sink.hasChanged("num")
+        sink.hasChanged()
+
+    }
+
+    void "PROBLEM - remains dirty untill flush"() {
```

Comment:
@snimavat this is the one we need to sort out. why does it show hasChanged until its flushed but that does not seem to be required on the first persist?